ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -574,49 +558,29 @@ loadSessionWithOptions recorder SessionLoadingOptions{..} dir = do
               -- . The information for the new component which caused this cache miss
               -- . The modified information (without -inplace flags) for
               --   existing packages
-              pure (Map.insert hieYaml (newHscEnv, NE.toList new_deps) m, (newHscEnv, NE.head new_deps', NE.tail new_deps'))
+              let (new,old) = NE.splitAt (NE.length new_deps) all_deps'
+              pure (Map.insert hieYaml (NE.toList all_deps) m, (new,old))
```

Comment:
The comment above is outdated now, I think.

Also, is `all_deps` correct here? Will we not remember too many ComponentInfos if we lazily load all components of a project?
I understand, we don't *load* all of them since we remove duplicates later on in `newComponentCache`, but it seems weird to me.

Reply:
I've updated the comment.

We remember RawComponentInfos corresponding to each file that caused a component to be loaded. If the options for each component list out all the targets in that component, then this `[RawComponentInfos]`(all_deps) will correspond exactly to the home units in the project.

The only we we have duplication is if the options for a component don't list all the targets. We will add a "special target" for the file that caused the component to be loaded in the first place. However, when a different file loaded, since the targets were not explicitly listed, we will create a new `RawComponentInfo` for that file, so we will have multiple `RawComponentInfos` for the same component, but loaded against different files. The duplicate merging logic in `newComponentCache` takes care of that.